src/system/interface.rs

```diff
@@ -1,14 +1,18 @@
 use std::{ffi::CStr, fmt::Display, num::ParseIntError, str::FromStr};
 
+#[repr(transparent)]
```

Comment:
`#[repr(transparent)]` is necessary in exactly two cases:

* You need the memory layout to be identical to that of the inner type due to unsafe code depending on this.
* You need the calling convention to be identical to that of the inner type due to using the transparent wrapper in the place of the inner type in function signatures.

Here you seem to have used to just for performance reasons. This is not exactly performance sensitive code and in practice the compiler will handle single field structs the same way as the inner type with respect to the memory layout and the rust calling convention (but not the C calling convention!), while this is not guaranteed, it should be fine to omit `#[repr(transparent)]` if performance is the only reason to have it.

Reply:
My train of thought was that one of the compliance tests failed because `ProcessId` was now a type wrapped in a struct, and `#[repr(transparent)]` would keep it the same size as `libc::pid_t`.

I could be wrong though. I'll try to debug it locally to see why it failed.

Reply:
Weird. It doesn't seem like the memory layout should affect compliance tests. IPC between different sudo processes does depend on memory layout, but because IPC is only done between forked processes, everyone should agree on the same memory layout and thus things should work out fine.

In any case I've left a suggestion for a change that would actually depend on `#[repr(transparent)]` for safety, so keeping it is fine IMO. Still it would be nice to know why the compliance test fails without it.

Reply:
I can't reproduce the problem locally and the test succeeded this time, so I guess the PR isn't the culprit.
Weird that it happened though, maybe someone else can reproduce or debug this?